lib/checksrc.pl

```diff
@@ -502,12 +502,19 @@ sub scanfile {
         }
         # check for '== NULL' in if/while conditions but not if the thing on
         # the left of it is a function call
-        if($nostr =~ /^(.*)(if|while)(\(.*[^)]) == NULL/) {
+        if($nostr =~ /^(.*)(if|while)(\(.*[^)]) [!=]= NULL/) {
             checkwarn("EQUALSNULL", $line,
                       length($1) + length($2) + length($3),
                       $file, $l, "we prefer !variable instead of \"== NULL\" comparisons");
         }
 
+        # check for 'NULL ==' in if/while conditions
+        if($nostr =~ /^(.*)(if|while)\(NULL [!=]=/) {
```

Comment:
I think you could combine these, and also it doesn't cover cases like (x) && y != NULL. I experimented in regex 101 and [this](https://regex101.com/r/xxEGWg/1) seems to work for all:
~~~perl
/^(.*)(if|while)(\(.*?)([!=]= NULL|NULL [!=]=)/
~~~

Reply:
Nice, thanks!